viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -0,0 +1,284 @@
+  }
+
+  /**
+   * Generating Line geometry and create the mesh.
+   * @param point Point from where the line will be generated.
+   */
+  private startLine(point: THREE.Vector3) {
+    this._linePosition[0] = this._linePosition[3] = point.x;
+    this._linePosition[1] = this._linePosition[4] = point.y;
+    this._linePosition[2] = this._linePosition[5] = point.z;
+
+    this._lineGeometry = new LineGeometry();
+    this._lineGeometry.setPositions(this._linePosition);
+
+    this._lineMaterial = new LineMaterial({ color: this._lineOptions.color, linewidth: this._lineOptions.lineWidth });
+
+    this._lineMesh = new THREE.Mesh(this._lineGeometry, this._lineMaterial);
+    this._viewer.addObject3D(this._lineMesh);
+  }
+
+  /**
+   * End the measurement.
+   * @param point Point at which measuring line ends.
+   */
+  private endMeasurement(point: THREE.Vector3) {
+    this.updateMeasurement(0, 0, point);
+
+    const labelPosition = this.calculateMidpoint(
+      new THREE.Vector3(this._linePosition[0], this._linePosition[1], this._linePosition[2]),
+      point
+    );
+    this.assignMeasurementValue();
+    this._measurementLabel.add(labelPosition, this._distanceValue);
+    this.clearLineObjects();
+  }
+
+  /**
+   * Calculate mid point betwen two given points.
+   * @param startPoint first point.
+   * @param endPoint second point.
+   * @returns Returns mid point between two points.
+   */
+  private calculateMidpoint(startPoint: THREE.Vector3, endPoint: THREE.Vector3): THREE.Vector3 {
+    let direction = endPoint.clone().sub(startPoint);
+    const length = direction.length();
+    direction = direction.normalize().multiplyScalar(length * 0.5);
+
+    return startPoint.clone().add(direction);
+  }
+
+  /**
+   * Get the distance between the measuring line start point & end point.
+   * @returns Return distance between start & end point of the line.
+   */
+  private getMeasurementValue() {
+    const startPoint = new THREE.Vector3(this._linePosition[0], this._linePosition[1], this._linePosition[2]);
+    const endPoint = new THREE.Vector3(this._linePosition[3], this._linePosition[4], this._linePosition[5]);
+    return endPoint.distanceTo(startPoint);
+  }
+
+  /**
+   * Check if their is a request to show the measurment with custom callback from the user.
+   */
+  private assignMeasurementValue() {
+    const options = this._options.changeMeasurementLabelMetrics;
+    if (options === undefined) {
+      this._distanceValue = this.getMeasurementValue().toFixed(2) + ' m';
+    } else {
+      this.updateMeasurementValue(options);
+    }
+  }
+
+  /**
+   * Update the measurement data.
+   * @param options Callback function which get user value to be added into label.
+   */
+  private updateMeasurementValue(options: MeasurementLabelUpdateDelegate) {
+    const measurementData = options(this.getMeasurementValue());
+    this._distanceValue = measurementData.distance.toFixed(2) + ' ' + measurementData.units;
+  }
```

Comment:
Instead of dealing with checking if this option is set, I recommend using a default callback that simply adds a "m"-suffix.

In MeasurementTool, store `options` as `Required<MeasurementToolOptions>` and merge provided options with default options on construction, e.g.
```
this._options = {...defaultMeasurmentToolOptions, ...options};
```

This is a much cleaner approach than checking for options set everywhere and defaulting to a default value when there is no option set.

Reply:
@larsmoa what about this
 `const measurementData = options(this.getMeasuredDistance()) || {
      distance: this.getMeasuredDistance(),
      units: 'm'
    };`